### DISCUSSION TITLE
Force 100% SOC by 5:30

### DISCUSSION THREAD

**Discussion by radiatorsimon** [springfall2008/batpred#1439] [General]
Any thoughts on how to force 100% SOC by 5:30 (end of cheap rate)? See attached screenshots showing Predbat only reaching 74% by 5:30 on the current plan. After 5:30 the import rate increases, and solar export is 15p/kwh so from a financial perspective I'm not sure why Predbat isn't charging to 100% so that any solar after 5:30 is just exported.

I've had a heat pump installed in the summer and want to ensure I've got 100% SOC at 5:30 to minimise heat pump running costs.

Any suggestions greatly appreciated. Happy to share additional files / logs as required.

![image](https://github.com/user-attachments/assets/32f7054f-50cd-4291-baa2-b367c00f35b6)
![image](https://github.com/user-attachments/assets/958c8cd9-74a1-4447-b1e0-2377e46d8af0)

**Comment by gcoan** [comment]
If you can turn debug mode on and then share the resultant HTML plan.  My guess is something to do with the charge/discharge/inverter loss figures that is causing Predbat to not think its worthwhile fully charging on the cheaper rate

**Reply by radiatorsimon** [reply]
Thanks Geoffrey, I've commented out the  solcast entities in apps.yaml and it's now working as desired.

**Reply by gcoan** [reply]
That would work but it means that predbat doesn't think you are getting any solar generation.

It should be predicting the load and solar forecast correctly to optimise your cost, it sounds like a fudge to cancel the solar generation.

**Comment by springfall2008** [comment]
Have you tried turning off 'calculate discharge during charge'?

**Reply by radiatorsimon** [reply]
Thanks both - now seems to be working correctly after turning off "Calculate Discharge on charge slots", will see what the SOC is in the morning. Have re-enabled Solcast too as whilst it worked, as Geoffrey noted it's not the nicest of solutions.

![image](https://github.com/user-attachments/assets/2d155f14-e508-45d8-8d1c-e0ae8c074836)
